internal/rsl/rsl.go

```diff
@@ -0,0 +1,296 @@
+		lines = append(lines, fmt.Sprintf("%s: false", SkipKey))
+	}
+
+	if len(a.Message) != 0 {
+		var message strings.Builder
+		messageBlock := pem.Block{
+			Type:  AnnotationMessageBlockType,
+			Bytes: []byte(a.Message),
+		}
+		if err := pem.Encode(&message, &messageBlock); err != nil {
+			return "", err
+		}
+		lines = append(lines, strings.TrimSpace(message.String()))
+	}
+
+	return strings.Join(lines, "\n"), nil
+}
+
+// GetLatestEntry returns the latest entry available locally in the RSL.
+// TODO: There is no information yet about the signature for the entry.
+func GetLatestEntry() (EntryType, error) {
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		return nil, err
+	}
+
+	ref, err := repo.Reference(plumbing.ReferenceName(RSLRef), true)
+	if err != nil {
+		return nil, err
+	}
+
+	commitObj, err := repo.CommitObject(ref.Hash())
+	if err != nil {
+		return nil, err
+	}
+
+	return parseRSLEntryText(commitObj.Message)
+}
+
+// GetEntry returns the entry corresponding to entryID.
+// TODO: There is no information yet about the signature for the entry.
+func GetEntry(entryID plumbing.Hash) (EntryType, error) {
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		return nil, err
+	}
+
+	ref, err := repo.Reference(plumbing.ReferenceName(RSLRef), true)
+	if err != nil {
+		return nil, err
+	}
+
+	iteratorHash := ref.Hash()
+	for {
+		commitObj, err := repo.CommitObject(iteratorHash)
+		if err != nil {
+			return nil, err
+		}
+
+		if iteratorHash == entryID {
+			return parseRSLEntryText(commitObj.Message)
+		}
+
+		if len(commitObj.ParentHashes) == 0 {
+			return nil, ErrRSLEntryNotFound
+		}
+
+		if len(commitObj.ParentHashes) > 1 {
+			return nil, ErrRSLBranchDetected
+		}
+
+		iteratorHash = commitObj.ParentHashes[0]
+	}
+}
+
+func parseRSLEntryText(text string) (EntryType, error) {
+	text = strings.TrimSpace(text)
+	if strings.HasPrefix(text, AnnotationHeader) {
+		return parseAnnotationText(text)
+	}
+	return parseEntryText(text)
+}
+
+func parseEntryText(text string) (*Entry, error) {
+	lines := strings.Split(text, "\n")
+	if len(lines) < 4 {
+		return nil, ErrInvalidRSLEntry
+	}
+	lines = lines[2:]
+
+	entry := &Entry{}
+	for _, l := range lines {
+		l = strings.TrimSpace(l)
+
+		ls := strings.Split(l, ":")
+		if len(ls) < 2 {
+			return nil, ErrInvalidRSLEntry
+		}
+
+		switch strings.TrimSpace(ls[0]) {
+		case RefKey:
+			entry.RefName = strings.TrimSpace(ls[1])
+		case CommitIDKey:
+			entry.CommitID = plumbing.NewHash(strings.TrimSpace(ls[1]))
+		}
+	}
+
+	return entry, nil
+}
+
+func parseAnnotationText(text string) (*Annotation, error) {
+	annotation := &Annotation{
+		RSLEntryIDs: []plumbing.Hash{},
+	}
+
+	messageBlock, _ := pem.Decode([]byte(text)) // rest doesn't seem to work when the PEM block is at the end of text, see: https://go.dev/play/p/oZysAfemA-v
+	if messageBlock != nil {
+		annotation.Message = string(messageBlock.Bytes)
+	}
+
+	lines := strings.Split(text, "\n")
+	lines = lines[2:]
```

Comment:
here as well!

Reply:
Same change made, < 4. I guess the test I added before wasn't actually triggered because of the missing header hehe.